src/pf-tooltip/pf-tooltip.component.js

```diff
@@ -0,0 +1,304 @@
+      this.setAttribute('container', value);
+    }
+  }
+
+  /**
+   * Get the animation duration
+   *
+   * @returns {string} The animation duration
+   */
+  get duration () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation duration
+   *
+   * @param {string} value The animation duration
+   */
+  set duration (value) {
+    if (this._duration !== value) {
+      this._duration = value;
+      this.setAttribute('duration', value);
+    }
+  }
+
+  /**
+   * Get the animation delay
+   *
+   * @returns {string} The animation delay
+   */
+  get delay () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation delay
+   *
+   * @param {string} value The animation delay
+   */
+  set delay (value) {
+    if (this._delay !== value) {
+      this._delay = value;
+      this.setAttribute('delay', value);
+    }
+  }
+
+  /**
+   * Get the placement position
+   *
+   * @returns {string} The placement position left, top, bottom, right
+   */
+  get placement () {
+    return this._placement;
+  }
+
+  /**
+   * Set placement position
+   *
+   * @param {string} value The placement position left, top, bottom, right
+   */
+  set placement (value) {
+    if (this._placement !== value) {
+      this._placement = value;
+      this.setAttribute('placement', value);
+    }
+  }
+
+  /**
+   * Get the target element
+   *
+   * @returns {string} The target element
+   */
+  get target () {
+    return this._target;
+  }
+
+  /**
+   * Set target element
+   *
+   * @param {string} value The target element
+   */
+  set target (value) {
+    if (this._target !== value) {
+      this._target = value;
+      this.setAttribute('target', value);
+    }
+  }
+
+  /**
+   * The open event handler
+   * @param e
+   */
+  open (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip === null) {
+        this.createTooltip();
+        this.styleTooltip();
+        this.showTooltip();
+      }
+    }, 20 );
+  }
+
+  /**
+   * The close event handler
+   * @param e
+   */
+  close (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip && this.tooltip !== null) {
+        pfUtil.removeClass(this.tooltip, 'in');
+        setTimeout(() => {
+          this.removeTooltip();
+        }, this._duration);
+      }
+    }, this._delay + this._duration);
+  }
+
+  /**
+   * Removes the tooltip
+   */
+  removeTooltip () {
+    this.tooltip && this.container.removeChild(this.tooltip);
+    this.tooltip = null;
+  }
+
+  /**
+   * Creates the tooltip
+   */
+  createTooltip () {
+    let clone = document.importNode(this._template.content, true);
+    let tooltipInner = clone.querySelector('.tooltip-inner');
+
+    //set tooltip content
+    tooltipInner.innerHTML = this.content;
+
+    //append to the container
+    this.container.appendChild(clone);
+
+    //set reference to appended node
+    let tooltips = this.container.querySelectorAll('.tooltip');
```

Comment:
The problem I was facing with this is `appendChild` does not return the actual DOM element reference (and instead returns the fragment). The idea here was to get the actual DOM reference after it was appended, that way it could be styled subsequently and referenced as `this.tooltip` after. Querying the last `.tooltip` element added to the page seemed to be the easiest way, but if we change the container or order that may not be the case. While this works (and I've tested hovering all of the different tooltips really quickly in different orders), there may be a better approach (such as querying the last `.tooltip` added within the container). Thoughts? Happy to research this further ...